packages/website/src/theme/MDXComponents/TryInPlayground.tsx

```diff
@@ -2,23 +2,45 @@ import Link from '@docusaurus/Link';
 import clsx from 'clsx';
 import React from 'react';
 
+import { fileTypes } from '../../components/options';
 import styles from './TryInPlayground.module.css';
 
+const fileExtensionsSortedByLength = fileTypes
+  .toSorted((a, b) => b.length - a.length)
+  .map(fileExtension => {
+    const language = fileExtension.replaceAll(/^\./g, '');
+    return [language, new RegExp(`^${language}\\b`)] as const;
+  });
+
 export function TryInPlayground({
   eslintrcHash,
   children,
   className,
   codeHash,
+  language,
 }: {
   eslintrcHash: string;
   children?: React.ReactNode;
   className?: string;
   codeHash?: string;
+  language?: string;
 }): React.ReactNode {
   const params = new URLSearchParams({ eslintrc: eslintrcHash });
   if (codeHash) {
     params.set('code', codeHash);
   }
+  if (language) {
+    // iterating over sorted array, so the longer extensions will be matched first
+    for (const [
+      fileExtension,
+      fileLanguageRegExp,
+    ] of fileExtensionsSortedByLength) {
+      if (fileLanguageRegExp.test(language)) {
+        params.set('fileType', `.${fileExtension}`);
+        break;
+      }
+    }
```

Comment:
Is it the same as this? or am I missing something?

```TS
const pattern = '('  + fileTypes.join('|') + ')$' ;
const regex = new RegExp(pattern);
const extension = language.match(regex)[1]
```

Reply:
Yeah, it's the same thing! What a cool solution!

But I personally like the current implementation better, as I find it a bit easier to read than RegExp :slightly_smiling_face: However, if others find the RegExp solution more readable, I'd be happy to use it!

Reply:
I personally find it easier to understand, but others might not think the same (since it's regex) 

Anyway, gotta admit that it took me couple of minutes to understand the logic here